api/capi/src/nntrainer.cpp

```diff
@@ -746,14 +743,36 @@ int ml_train_layer_destroy(ml_train_layer_h layer) {
   return status;
 }
 
-int ml_train_layer_set_property(ml_train_layer_h layer, ...) {
+static int nntrainer_layer_set_property(ml_train_layer_h layer,
+                                        std::vector<std::string> &param_list) {
   int status = ML_ERROR_NONE;
   ml_train_layer *nnlayer;
   const char *data;
   std::shared_ptr<ml::train::Layer> l;
 
-  check_feature_state();
+  ML_TRAIN_VERIFY_VALID_HANDLE(layer);
+
+  {
+    ML_TRAIN_GET_VALID_LAYER_LOCKED(nnlayer, layer);
+    ML_TRAIN_ADOPT_LOCK(nnlayer, layer_lock);
+
+    l = nnlayer->layer;
+  }
 
+  returnable f = [&]() {
+    l->setProperty(param_list);
+    return ML_ERROR_NONE;
+  };
+  status = nntrainer_exception_boundary(f);
+
+  return status;
+}
+
+int ml_train_layer_set_property(ml_train_layer_h layer, ...) {
+  int status = ML_ERROR_NONE;
+  const char *data;
+
+  check_feature_state();
```

Comment:
We can do this inside of nntrainer_layer_set_property at line 746. Then we can delete it at line 775 and line 795.